test/gtest/common/iota_view.cpp

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+
+template <typename DistVecT> class IotaView : public testing::Test {
+public:
+};
+
+TYPED_TEST_SUITE(IotaView, AllTypes);
+
+TYPED_TEST(IotaView, ZipWithDR) {
+  xhp::distributed_vector<int> dv(20);
+  auto v = dr::views::iota(1, 20);
+  int ref = 1;
+
+  auto z = xhp::views::zip(dv, v);
+  for (auto itr = z.begin(); itr != z.end(); itr++) {
+    auto [dve, ve] = *itr;
+    dve = ve;
+    EXPECT_EQ(dv[ref - 1], ref);
+    ref++;
+  }
+}
+
+TYPED_TEST(IotaView, Copy) {
+  TypeParam dv(10);
+  auto v = dr::views::iota(1, 11);
+
+  rng::copy(v, dv.begin());
```

Comment:
```suggestion
  xhp::copy(v, dv.begin());
```

And same for other algorithms

Reply:
dr:views::iota_view has only rng::views::iota below it
As it is not a view at real data, it has no segments. We can:
- change the requirements for input in algorithms (output will be distributed_iterator anyway); for dr::mhp::transform it is enough, because it has dr::mhp::zip inside, and execution is distributed anyway, not sure what to do with other algorithms
- do a segments() function to meet the distributed_range requirement; as we have no physical storage, what should it return?

Reply:
I prefer the first option. MHP has a SPMD programming model, and my thinking is that algorithms can assume that arguments that are not distributed_ranges are replicated across the processes and can be segmented as desired. Zip does that, but I only tested for iota. This is not what you want for scatter/gather where it is an all-to-1 or 1-to-all communication (e.g. distribute a local vector resident on a single rank). For that case, I have overloads that also accept a root rank.

@BenBrock: I know that SHP zip also handles iota. What is the general model?

Reply:
Made copy() specialized for dr::views::iota in [fd4ca86](https://github.com/oneapi-src/distributed-ranges/pull/300/commits/fd4ca869190e32fa75ce9e73edd3a23ad8cc47b6). As iota view is generated, not stored, it required scattering subsequent values from root node to distributed segments.